core/src/avm1/globals/movie_clip.rs

```diff
@@ -121,6 +122,62 @@ pub fn constructor<'gc>(
     Ok(this.into())
 }
 
+fn new_rectangle<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    rectangle: Rectangle,
+) -> Result<Value<'gc>, Error<'gc>> {
+    let x = rectangle.x_min.to_pixels();
+    let y = rectangle.y_min.to_pixels();
+    let width = (rectangle.x_max - rectangle.x_min).to_pixels();
+    let height = (rectangle.y_max - rectangle.y_min).to_pixels();
+    let args = &[x.into(), y.into(), width.into(), height.into()];
+    let proto = activation.context.avm1.prototypes.rectangle;
+    proto.construct(activation, args)
+}
+
+fn object_to_rectangle<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    object: Object<'gc>,
+) -> Result<Rectangle, Error<'gc>> {
+    let x = object.get("x", activation)?.coerce_to_i32(activation)?;
+    let y = object.get("y", activation)?.coerce_to_i32(activation)?;
+    let width = object.get("width", activation)?.coerce_to_i32(activation)?;
+    let height = object
+        .get("height", activation)?
+        .coerce_to_i32(activation)?;
+    Ok(Rectangle {
+        x_min: Twips::from_pixels_i32(x),
+        x_max: Twips::from_pixels_i32(x + width),
+        y_min: Twips::from_pixels_i32(y),
+        y_max: Twips::from_pixels_i32(y + height),
+    })
+}
+
+fn scroll_rect<'gc>(
+    this: MovieClip<'gc>,
+    activation: &mut Activation<'_, 'gc, '_>,
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(scroll_rect) = this.next_scroll_rect() {
+        new_rectangle(activation, scroll_rect)
+    } else {
+        Ok(Value::Undefined)
+    }
+}
+
+fn set_scroll_rect<'gc>(
+    this: MovieClip<'gc>,
+    activation: &mut Activation<'_, 'gc, '_>,
+    value: Value<'gc>,
+) -> Result<(), Error<'gc>> {
+    let rectangle = if let Value::Object(object) = value {
```

Comment:
Looks like this should not set if the object is a display object as well (yet another case of this).

```
_root.x = 0;
_root.y = 0;
_root.width = 10;
_root.height = 10;
test.scrollRect = _root;
trace(test.scrollRect); // undefined
```

Reply:
Again, another testimony for `StageObject` being a separate `Value` member, and not an `Object`.

Reply:
I do not indend to resolve this in current PR since I believe this is a codebase-wide issue.